### PR TITLE
 ENH: Add `is_integer` to `np.floating` & `np.integer` 

### PR DESCRIPTION
**Pull request by BvB93**
The `builtins.float` class currently contains a number of methods that are absent from its numpy-based
counterparts (xref https://github.com/numpy/numpy/issues/13375). This PR alleviates this issue for `is_integer`, which is now available to 
`np.floating` and `np.integer` subclasses, matching builtin python.

### CODE REVIEW

File: numpy/core/src/multiarray/scalartypes.c.src [hunk 1]
```diff
@@ -1908,6 +1908,34 @@ error:
...
+        Py_RETURN_FALSE;
+    }
+    if (!npy_isfinite(val)) {
+        Py_RETURN_FALSE;
+    }
+
+    o = (npy_floor@c@(val) == val) ? Py_True : Py_False;
+    Py_INCREF(o);
+    return o;
+}
```
Comment (BvB93) [inline numpy/core/src/multiarray/scalartypes.c.src:1937]: The implementation is largly based on its counterpart in cpython: https://github.com/python/cpython/blob/60b93d9e4922eeae25052bc15909d1f4152babde/Objects/floatobject.c#L853-L873

Comment (charris) [reply numpy/core/src/multiarray/scalartypes.c.src:1937]: Maybe `res` instead of `o`?

Comment (BvB93) [reply numpy/core/src/multiarray/scalartypes.c.src]: Sure, why not. Done as of https://github.com/numpy/numpy/pull/19803/commits/bb3e4077a32078deb31d7c31034303303fc268f4.

**Comment by charris** [comment]
Thanks Bas.

**Pull request merged**